File: Carts/Animation/View Controller Presentation/CAPullToCloseTransitionPresentationController.m
//
//  CAPullToCloseTransitionPresentationController.m
//  GroceryList
//
//  Created by Hakon Hanesand on 4/6/15.

#import "CAPullToCloseTransitionPresentationController.h"

@implementation CAPullToCloseTransitionPresentationController

- (void)presentationTransitionWillBegin {
    //add the presented view to the heirarchy
    [[self containerView] addSubview:[self presentedView]];
}

- (void)presentationTransitionDidEnd:(BOOL)completed {
    //nothing to do here
}

- (void)dismissalTransitionWillBegin {
    //again, nothing needs to be done here
}

- (void)dismissalTransitionDidEnd:(BOOL)completed {
    //nothing here
}

- (CGRect)frameOfPresentedViewInContainerView {
    //return the full size of the screen (ie the entire container view)
    return [self containerView].bounds;
}

@end

File: Carts/Animation/View Controller Presentation/CAPullToCloseTransitionPresentationController.swift
//
//  CAPullToCloseTransitionPresentationController.swift
//  Carts
//
//  Created by Hakon Hanesand on 4/26/15.
//  Copyright (c) 2015 Hakon Hanesand. All rights reserved.
//

import UIKit

class CAPullToCloseTransitionPresentationController: UIPresentationController {

    override func presentationTransitionWillBegin() {
        super.presentationTransitionWillBegin()
        self.containerView!.addSubview(self.presentedView()!)
    }
    
    override func presentationTransitionDidEnd(completed: Bool) {
        super.presentationTransitionDidEnd(completed)
    }
    
    override func dismissalTransitionWillBegin() {
        super.dismissalTransitionWillBegin()
    }
    
    override func dismissalTransitionDidEnd(completed: Bool) {
        super.dismissalTransitionDidEnd(completed)
    }
    
    override func frameOfPresentedViewInContainerView() -> CGRect {
        return self.containerView!.bounds
    }
}
